mono/metadata/marshal.c

```diff
@@ -990,7 +990,8 @@ mono_string_builder_to_utf8 (MonoStringBuilder *sb)
  *
  * Returns: a utf16 string with the contents of the StringBuilder.
  *
- * The return value must not be freed.
+ * The return value must be released with mono_marshal_free.
```

Comment:
Possibly yes. I don't have the big picture here, but I could only find one place where the `need_free` result from `mono_marshal_need_free` is actually used and that's in `emit_marshal_array`: 
[marshal.c#L6357] 
(https://github.com/mono/mono/blob/bea97b9d05fbb675bce541647d77b7c15d41fe03/mono/metadata/marshal.c#L6357)